src/scenes/ProgressReports/EditForm/Steps/SubmitReportStep/ConfirmVariantDialog.tsx

```diff
@@ -0,0 +1,142 @@
+import {
+  Button,
+  Dialog,
+  DialogActions,
+  DialogContent,
+  DialogTitle,
+  Typography,
+} from '@mui/material';
+import { Role } from '~/api/schema/schema';
+import { PromptResponseListFragment } from '~/common/fragments';
+import { useSession } from '~/components/Session';
+import { ProgressReportEditFragment } from '../../ProgressReportEdit.graphql';
+
+interface ConfirmEmptyVariantsProps {
+  report: ProgressReportEditFragment;
+  open: boolean;
+  onContinue: () => void;
+  onCancel: () => void;
+  updateConfirmStatus: () => void;
+}
+
+export const ConfirmEmptyVariants = ({
+  report,
+  open,
+  onContinue,
+  onCancel,
+  updateConfirmStatus,
+}: ConfirmEmptyVariantsProps) => {
+  const { reportFile, teamNews, communityStories } = report;
+  const { session } = useSession();
+  const roles = new Set(session?.roles.value);
+  const hasReportProgressFile = !!reportFile.value;
+  const projectMgmtHeader = !hasReportProgressFile && reportFile.canEdit;
+  const hasSelectedTeamNewsPrompt = !!(teamNews.items.length > 0);
+  const hasSelectedStoryPrompt = !!(communityStories.items.length > 0);
+  const listItems = [];
+
+  const handleConfirmDialog = () => {
+    onContinue();
+  };
+
+  const handleCloseDialog = () => {
+    onCancel();
+  };
+
+  const getEmptyResponsesForRole = (
+    list: PromptResponseListFragment,
+    roles: Set<Role>
+  ) => {
+    if (list.items.length === 0) return [];
+    const responsesForRole = list.items[0]?.responses.filter((item) => {
+      return item.variant.responsibleRole
+        ? roles.has(item.variant.responsibleRole)
+        : false;
+    });
+    const emptyResponsesByRole = responsesForRole?.filter((item) =>
+      !item.response.value && item.response.canEdit ? true : false
+    );
+    return emptyResponsesByRole ? emptyResponsesByRole : [];
+  };
+
+  const emptyTeamNewsResponsesByRole = getEmptyResponsesForRole(
+    teamNews,
+    roles
+  );
+  if (!hasSelectedTeamNewsPrompt || emptyTeamNewsResponsesByRole.length > 0) {
+    listItems.push(<li>Team News</li>);
+  }
+
+  const emptyStoryResponsesByRole = getEmptyResponsesForRole(
+    communityStories,
+    roles
+  );
+  if (!hasSelectedStoryPrompt || emptyStoryResponsesByRole.length > 0) {
+    listItems.push(<li>Story</li>);
+  }
+
+  const investorConnectionHeader = !!(listItems.length > 0);
+  if (!investorConnectionHeader && !projectMgmtHeader && !open) {
+    updateConfirmStatus();
```

Comment:
Nit: consider renaming this to 'onUpdateConfirmStatus' (callbacks that act on an action are commonly prefixed with 'on' in react)